packages/Takeover/src/components/Header.js

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import { fontSize, spacer } from "@paprika/stylers/lib/helpers";
+import tokens from "@paprika/tokens/lib/tokens";
+import Button from "@paprika/button";
+
+const propTypes = {
+  children: PropTypes.node.isRequired,
+  hasCloseButton: PropTypes.bool,
+  kind: PropTypes.oneOf(["default", "primary"]),
+  onClose: PropTypes.func,
+};
+
+const defaultProps = {
+  hasCloseButton: true,
+  kind: "default",
+  onClose: () => {},
+};
+
+const kind = {
+  default: `background: ${tokens.color.white}; color: ${tokens.color.black};`,
+  primary: `background: ${tokens.color.purple}; color: ${tokens.color.white};`,
+};
+
+const Wrapper = styled.div`
+  align-items: center;
+  border-bottom: 1px solid ${tokens.border.color};
+  box-shadow: ${tokens.shadow};
+  box-sizing: border-box;
+  display: flex;
+  height: ${spacer(6)};
+  justify-content: space-between;
+  min-height: ${spacer(6)};
+  padding: 0 ${spacer(2)};
+  width: 100%;
+
+  &:focus {
+    outline: 0;
+  }
+
+  .heading--level-1,
+  .heading--level-2,
+  .heading--level-3,
+  .heading--level-4,
+  .heading--level-5 {
+    ${fontSize()};
+    font-weight: 400;
+    margin: 0;
+  }
+
+  ${props => kind[props.kind]}
+`;
+
+const StyledCloseButton = styled(Button.Close)`
+  .button__icon {
```

Comment:
I would actually like to get rid of this class in the `<Button>` (and avoid classes as much as possible in Paprika components).  Instead I suggest using the `[data-pka-anchor='button.icon']` selector.

Reply:
In my opinion, the best option is to convert [`ButtonIcon` component](https://github.com/acl-services/paprika/blob/eb1ee6765fce360df1d296e2ffa1c9d0e417a917/packages/Button/src/Button.js#L96) to styled component:

```jsx
export const ButtonIcon = styled(props => props.children ? <span {...props} /> : null)`
  ${iconStyles};
`;
```


And export `ButtonIcon` const from `Button.js`:

```jsx
...

export const ButtonIcon = ...;

const Button = ...;

export default Button;
```

And then style icon in place in following way:
```jsx
import { ButtonIcon } from '@paprika/button';

const StyledCloseButton = styled(Button.Close)`
  ${ButtonIcon} {
    ...
  }
`;
```

Reply:
Agreed. I had a similar conversation recently with Tristan, and we came to the same conclusion.